automatminer/automl/adaptors.py

```diff
@@ -226,6 +228,129 @@ def predict(self, df, target):
+                of training examples per epoch
+            activation: Activation function to use for neural network. By default, we use relu
+            input_noise: Any noise that must be added to the input of the system
+            use_maxout: Boolean to indicate if network should be maxout network
+            use_maxnorm: Boolean to indicate if maxnorm regularization constraints should be used
+            learning_rate: Specifies learning rate decay of ADAM optimizer
+            stop_early: True if neural network should stop training after loss starts to increases, false otherwise.
+            **kwargs: Additional (optional) args needed for neural network wrapper.
+
+        Attributes:
+            The following attributes are set during fitting.
+
+            mode (str): Either "regression" or "classification"
+            features (list): The features labels used to develop the ml model.
+            ml_data (dict): The raw ml data used for training.
+            backend (NnWrapper): The Keras neural network interface used for ML training.
+            is_fit (bool): If True, the adaptor and backend are fit to a dataset.
+            fitted_target (str): The target name in the df used for training.
+    """
+    def __init__(self, logger=True, hidden_layer_sizes=(100,), dropout=0.5, show_accuracy=True, batch_spec=((400, 1024), (100, -1)), activation="relu", input_noise=0., use_maxout=False, use_maxnorm=False, learning_rate=0.001, stop_early=False, **kwargs):
+        self.mode = None
+        self.hidden_layer_sizes = hidden_layer_sizes
+        self.dropout = dropout
+        self.show_accuracy = show_accuracy
+        self.batch_spec = batch_spec
+        self.activation = activation
+        self.input_noise = input_noise
+        self.use_maxout = use_maxout
+        self.use_maxnorm = use_maxnorm
+        self.learning_rate = learning_rate
+        self.stop_early = stop_early
+        self._backend = None
+        self.kwargs = kwargs
+        self._features = None
+        self._logger = self.get_logger(logger)
+        self.is_fit = False
+
+    @set_fitted
+    def fit(self, df, target, **fit_kwargs):
+        """
+        Train a neural network by fitting on a dataframe.
+
+        Args:
+            df (pandas.DataFrame): The df to be used for training.
+            target (str): The key used to identify the machine learning target.
+            **fit_kwargs: Keyword arguments to be passed to the neural network's fit method.
+                These arguments must be valid arguments to the Keras neural network fit method.
+
+        Returns:
+            NeuralNetworkAdaptor(self)
+
+        """
+        y = df[target]
+        X = df.drop(columns=target)
+
+        # Determine learning type based on whether classification or regression
+        self.mode = regression_or_classification(df[target])
+        if self.mode == "classification" or self.mode == "regression":
```

Comment:
this "if" check is not needed. `regression_or_classification" will return either regression or classification, or throw an error otherwise.